mlflow/store/tracking/rest_store.py

```diff
@@ -356,6 +362,27 @@ def start_trace(self, trace_info: TraceInfo) -> TraceInfo:
         Returns:
             The returned TraceInfo object from the backend.
         """
+        try:
+            sql_warehouse_id = MLFLOW_TRACING_SQL_WAREHOUSE_ID.get()
+            req_body = message_to_json(
+                CreateTrace(trace_info=trace_info.to_proto_v4(), sql_warehouse_id=sql_warehouse_id)
```

Comment:
qq: how does this handler work if `MLFLOW_TRACING_SQL_WAREHOUSE_ID` is not set? We'll have some backend logic to retrieve a "default SQL warehouse" if one is not provided, so we do want this to run even if it is not set